test/js/web/fetch/fetch-readablestream-request-body.test.ts

```diff
@@ -0,0 +1,434 @@
+import type { DirectUnderlyingSource } from "bun";
+import { describe, expect, test } from "bun:test";
+
+describe("fetch with Request body lifecycle", () => {
+  test("should properly handle Request with streaming body", async () => {
+    using server = Bun.serve({
+      port: 0,
+      async fetch(req) {
+        const text = await req.text();
+        return new Response(text);
+      },
+    });
+
+    const chunk = new TextEncoder().encode("test data");
+
+    const originalRequest = new Request(server.url, {
+      method: "POST",
+      body: new ReadableStream({
+        async start(controller) {
```

Comment:
This test should use `async pull` with a delay of say 0ms because otherwise we may detect that the ReadableStream has already completed by the time the `fetch` callback is called.